### PR TITLE
NPU Piecewise Graph

### PR DESCRIPTION
**Pull request by eshoguli**
## Motivation
Use command line parameters to enable and get max performance:
```
--enable-piecewise-graph --cuda-graph-bs 128 --cuda-graph-max-bs 128
```

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/entrypoints/http_server_engine.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def launch_server_process(server_args: ServerArgs) -> multiprocessing.Process:
     p.start()
 
     base_url = server_args.url()
-    timeout = 300.0  # Increased timeout to 5 minutes for downloading large models
```
Comment (ssshinigami) [inline python/sglang/srt/entrypoints/http_server_engine.py:26]: please remove it, nobody agree to change it

File: python/sglang/srt/layers/attention/base_attn_backend.py [hunk 2]
```diff
@@ -5,9 +5,10 @@
 
 import torch
 
+from sglang.srt.model_executor.forward_batch_info import ForwardMode
```
Comment (ssshinigami) [inline python/sglang/srt/layers/attention/base_attn_backend.py:8]: what is a reason of this changes?

File: python/sglang/srt/layers/logits_processor.py [hunk 3]
```diff
@@ -122,9 +122,9 @@ class LogitsMetadata:
     @classmethod
     def from_forward_batch(cls, forward_batch: ForwardBatch):
         if (
-            forward_batch.forward_mode.is_extend()
```
Comment (ssshinigami) [inline python/sglang/srt/layers/logits_processor.py:125]: please keep it as before

File: python/sglang/srt/managers/schedule_batch.py [hunk 4]
```diff
@@ -1340,6 +1343,10 @@ def prepare_for_extend(self):
             self.model_config.vocab_size,
         )
 
+        # pid = os.getpid()
```
Comment (ssshinigami) [inline python/sglang/srt/managers/schedule_batch.py:1346]: please remove all comments

File: python/sglang/srt/model_executor/compilation/npu_compiler_backend.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    get_pp_group,
+    get_tensor_model_parallel_rank,
+    get_tensor_model_parallel_world_size,
+)
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class submod_1(torch.nn.Module):
```
Comment (ssshinigami) [inline python/sglang/srt/model_executor/compilation/npu_compiler_backend.py:38]: please rename it

File: python/sglang/srt/model_executor/compilation/compilation_context.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import torch_npu
+
+class CompilationContext:
+    graph_memory_pool = None
+    stream: torch_npu.npu.Stream = None
```
Comment (ssshinigami) [inline python/sglang/srt/model_executor/compilation/compilation_context.py:19]: please add white space

File: python/sglang/srt/model_executor/model_runner.py [hunk 7]
```diff
@@ -1416,6 +1426,29 @@ def init_attention_backend(self):
...
+
+        tic = time.perf_counter()
+        before_mem = get_available_gpu_memory(self.device, self.gpu_id)
+        logger.info(
+            f"Capture piecewise graph begin. This can take up to several minutes. avail mem={before_mem:.2f} GB"
+        )
+        self.graph_runner = PiecewiseGraphRunner(self, self.server_args.compilation_config)
+        after_mem = get_available_gpu_memory(self.device, self.gpu_id)
+        logger.info(
+            f"Capture piecewise graph end. Time elapsed: {time.perf_counter() - tic:.2f} s. "
```
Comment (ssshinigami) [inline python/sglang/srt/model_executor/model_runner.py:1448]: please remove all log info

File: python/sglang/srt/managers/schedule_batch.py [hunk 8]
```diff
@@ -40,6 +40,9 @@
 from itertools import chain
 from typing import TYPE_CHECKING, Any, List, Optional, Set, Tuple, Union
 
+import os
+import threading
```
Comment (ssshinigami) [inline python/sglang/srt/managers/schedule_batch.py:44]: please remove

**Review by ssshinigami — CHANGES_REQUESTED** [review]
